### PR TITLE
cmd,query: bump promql-engine and make subqueries for thanos engine configurable

### PR DESCRIPTION
**Pull request by MichaHoffmann**
* [x] I added CHANGELOG entry for this change.
* [ ] Change is not relevant to the end user.

## Changes

Bumped promql-engine; fixed fallout from the bump; added a configuration flag for enabling subqueries in the thanos engine

## Verification

Not tested yet.

### CODE REVIEW

File: cmd/thanos/query.go [hunk 1]
```diff
@@ -690,7 +696,11 @@ func runQuery(
 	}
 
 	engineFactory := apiv1.NewQueryEngineFactory(
-		engineOpts,
+		engine.Opts{
+			EngineOpts:       engineOpts,
+			Engine:           promql.NewEngine(engineOpts),
+			EnableSubqueries: enableThanosEngineSubqueries,
```
Comment (yeya24) [inline cmd/thanos/query.go:702]: Feel like it is an overkill to me to have a flag for it

Comment (MichaHoffmann) [reply cmd/thanos/query.go:702]: I added it by popular request to have this feature configurable ( for a short time ). Once this has proven stable enough we can probably remove the flag again.

Comment (MichaHoffmann) [reply cmd/thanos/query.go:702]: Having said that im fairly confident that it works like in upstream promql since it passes acceptance tests and fuzzing

Comment (GiedriusS) [reply cmd/thanos/query.go:702]: :heavy_plus_sign: maybe we can just enable subqueries? The more people exposed to this, the faster bugs will be fixed (if any)

Comment (MichaHoffmann) [reply cmd/thanos/query.go:702]: Can we have it for a short time and we run a nightly build and once we are happy ill take out the switch again?

Comment (MichaHoffmann) [reply cmd/thanos/query.go:702]: I guess we could use prometheus engine in case of issues too, so maybe its not so bad.

Comment (MichaHoffmann) [reply cmd/thanos/query.go:702]: Consulted with a teammate; ill take out the flag in a couple hours!

**Comment by yeya24** [comment]
Can we take a look at the failed unit test? https://app.circleci.com/pipelines/github/MichaHoffmann/thanos/79/workflows/475701d9-0860-424d-8f75-ccb828384599/jobs/83

**Comment by MichaHoffmann** [comment]
> Can we take a look at the failed unit test? https://app.circleci.com/pipelines/github/MichaHoffmann/thanos/79/workflows/475701d9-0860-424d-8f75-ccb828384599/jobs/83

Oh thats maybe related to the prometheus version bump and looks a bit scary

**Pull request closed without merging**